### ISSUE TITLE
Implement Phoenix.Socket in LFE

### ISSUE THREAD

**Issue by oubiwann** [lfe-dragon/dragon#1]

**Comment by John-Goff** [comment]
What is the goal here? To have a `Phoenix.Socket` compatible module written in LFE and used in a Phoenix project? To have a `Phoenix.Socket` written in Elixir usable in Dragon? To just copy the API or experience of `Phoenix.Socket` and make a "Lisp-y" version from scratch? Something else? Can't really figure out how to suggest ideas until I have a better idea of the direction.

**Comment by oubiwann** [comment]
Yeah, the over all goal of this collection of features (socket, channel, pubsub, etc.) is to be able to build LFE web apps that are messaging-aware, can send and receive to-and-from the browser, etc., powering all sorts of async web apps (my thought was to start with a chat app demo).

The technical goal isn't to re-use any Elixir code -- in fact, just for this work, I'd consider importing Elixir as an anti-pattern. Should be 100% LFE, as simple and clean as possible, etc.

That being said, outside the context of this particular effort, it would be amazing (in general) to have ready mechanisms that supported the (re)use of Elixir code in LFE projects.
